fido/Crypto/Fido2/PublicKey.hs

```diff
@@ -156,6 +153,20 @@ decodePublicKey = do
       key' <- CBOR.decodeIntCanonical
       when (mapKeyToInt key /= key') $ fail $ "Expected " ++ show key
 
+toCurveName :: COSEAlgorithmIdentifier -> ECC.CurveName
+toCurveName COSEAlgorithmIdentifierES256 = ECC.SEC_p256r1
+toCurveName COSEAlgorithmIdentifierES384 = ECC.SEC_p384r1
+toCurveName COSEAlgorithmIdentifierES512 = ECC.SEC_p521r1
+toCurveName _ = error "No known curve"
+
+mapKeyToInt :: MapKey -> Int
+mapKeyToInt key = case key of
+  Kty -> 1
```

Comment:
Is there a spec for these numbers we can link to? Also applies to other magic numbers in this file